lib/sensors/src/MoistureSensor.cpp

```diff
@@ -63,18 +62,19 @@ uint16_t MoistureSensor::Read_Moisture() {
     ThisThread::sleep_for(1s);
     i2c_.read(Sensor_I2C_Address, buf, 2);  // read moisture
 
-    ret = ((uint16_t)buf[0] << 8 | buf[1]);  // concatenate bytes together
+    sensorReading = ((uint16_t)buf[0] << 8 | buf[1]);  // concatenate bytes together
 
     counter--;
-  } while (ret == 65535 && counter != 0);  // repeat until value has been measured, or until loop has run 10 times
+  } while (sensorReading == 65535 && counter != 0);  // repeat until value has been measured, or until loop has run 10 times
                                            // (breaks out regardless of if read works or not)
 
-  return ret;
+  return true;
```

Comment:
Should it return false if counter reaches 0 and all read attempts were unsuccessful?

Reply:
From the previous comment I'll assume that if it fails sensorReading is 65535 so this should return false if the read fails.